overwatch/src/services/runner.rs

```diff
@@ -0,0 +1,350 @@
+                        info!("Service is already running.");
+                    } else {
+                        Self::handle_start::<Service>(
+                            &mut service_resources,
+                            &mut service_task_handle,
+                            &mut state_handle_task_handle,
+                        );
+                        service_lifecycle_phase = LifecyclePhase::Started;
+                    }
+
+                    // TODO: Sending a different signal could be handy to differentiate whether
+                    //  the service was already started or not.
+                    if let Err(error) = finished_signal_sender.send(()) {
+                        dbg!(
+                            "Error while sending the LifecycleMessage::Start signal: {}.",
+                            error
+                        );
+                    }
+                }
+                LifecycleMessage::Stop(finished_signal_sender) => {
+                    if service_lifecycle_phase == LifecyclePhase::Stopped {
+                        info!("Service is already stopped.");
+                    } else {
+                        Self::handle_stop(
+                            &mut service_task_handle,
+                            &mut state_handle_task_handle,
+                            &mut service_resources,
+                        )
+                        .await;
+                        service_lifecycle_phase = LifecyclePhase::Stopped;
+                    }
+
+                    // TODO: Sending a different signal could be handy to differentiate whether
+                    //  the service was already stopped or not.
+                    if let Err(error) = finished_signal_sender.send(()) {
+                        dbg!("Error while sending the LifecycleMessage::Stop finished signal: {}. Likely due to the receiver being already dropped in the Service::run task.", error);
+                    }
+                }
+            }
+        }
+    }
+
+    fn handle_start<Service>(
+        service_resources: &mut ServiceResources<
+            Message,
+            Settings,
+            State,
+            StateOp,
+            RuntimeServiceId,
+        >,
+        service_task_handle: &mut Option<JoinHandle<()>>,
+        state_handle_task_handle: &mut Option<JoinHandle<()>>,
+    ) where
+        Service: ServiceCore<RuntimeServiceId, Settings = Settings, State = State, Message = Message>
+            + 'static,
+        StateOp: Clone,
+    {
+        let initial_state = match service_resources.get_service_initial_state() {
+            Ok(initial_state) => initial_state,
+            Err(error) => {
+                panic!("Failed to create the initial state from settings: {error}");
+            }
+        };
+
+        let inbound_relay = service_resources
+            .inbound_relay
+            .take()
+            .expect("Failed to retrieve inbound relay.");
```

Comment:
What does it mean when this happens? Can we provide more info in the message?

Reply:
This is improved in one of the following PRs: It's encapsulated into `ServicesResources` and logs the actual error.